### ISSUE TITLE
RequestsSignatureDelegatingHandler is incompatible with IHttpClientFactory

### ISSUE THREAD

**Issue by jeremiethibeault** [nventive/RequestsSignature#2]
## Expected Behavior
 `RequestsSignatureDelegatingHandler` can be created without a `InnerHandler` to support `IHttpClientFactory`.

## Actual Behavior
If not provided, `RequestsSignatureDelegatingHandler` creates its own `InnerHandler`. When used with `IHttpClientFactory`, the expectation is that the `InnerHandler` is not set.

## Steps to Reproduce the Problem

1. Create an `HttpClient` using the `IHttpClientFactory`.
```csharp

serviceCollection.AddHttpClient("MyHttpClient")
  .AddHttpMessageHandler(() => new RequestsSignatureDelegatingHandler(new RequestsSignatureOptions()));

httpClientFactory.CreateClient("MyHttpClient");
```

2. Run and get the following exception:
```
System.InvalidOperationException: 'The 'InnerHandler' property must be null. 'DelegatingHandler' instances provided to 'HttpMessageHandlerBuilder' must not be reused or cached.
Handler: 'RequestsSignature.HttpClient.RequestsSignatureDelegatingHandler''
```

## Specifications

- Version: 2.0.0
- Platform: netstandard2.0

**Comment by julienblin** [comment]
I pushed a pull request and merged it in master.
This should be available really soon **in preview** with the following version number: [2.0.1-beta0001](https://www.nuget.org/packages/RequestsSignature.HttpClient/2.0.1-beta0001).

@jeremiethibeault - Could you please try with this version and let me know? If this is good I'll release a new official package.

Thanks!

**Comment by jeremiethibeault** [comment]
@julienblin Works as expected. Thanks!

**Comment by julienblin** [comment]
Released in [v3.0.0](https://github.com/nventive/RequestsSignature/blob/master/CHANGELOG.md#300).
